Tests/Unit/MSFT_xWebApplication.Tests.ps1

```diff
@@ -495,10 +495,26 @@ try
         Describe "how $script:DSCResourceName\Set-TargetResource responds to Ensure = 'Present'" {   
             
             Context 'Web Application does not exist' {
+
+                $script:mockGetWebApplicationCalled = 0
```

Comment:
You'll want to line up the tabs a little more here (nit picky I know but we're really pushing the style hard ont this resource.

````powershell
$mockWebApplication = {
    $script:mockGetWebApplicationCalled++
    if($script:mockGetWebApplicationCalled -eq 1)
    {
        return $null
    }
    else
    {
        return @{
            ApplicationPool = $MockParameters.WebAppPool
            PhysicalPath    = $MockParameters.PhysicalPath
            ItemXPath       = ("/system.applicationHost/sites/site[@name='{0}']/application[@path='/{1}']" -f $MockParameters.Website, $MockParameters.Name)
            Count           = 1
        }
    }
}
````